google/api/annotations.proto

```diff
@@ -102,3 +102,21 @@ extend google.protobuf.MethodOptions {
   // See `google/api/http.proto`.
   HttpRule http = 72295728;
 }
+
+
+// Common Resource Imports. (See `resources.proto` in this directory.)
```

Comment:
I think these would be better in resources.proto.

Reply:
Me too, though we would have to declare them as messages instead of as a resource_definition option (which would require a circular import between annotations.proto, which contains the resource_definition option, and this resources.proto file). 

Another option is to put it in a completely separate file that imports annotations.proto and resources.proto; maybe call it "common_resources.proto". Thoughts?

Reply:
+1 on common_resources.proto

Reply:
I also would prefer these to be in a separate file. `common_resources.proto` sgtm. @lukesneeringer what do you think?

But we should probably update the proto linter to check that all resource-name dependencies have been imported property, because protoc won't complain. A micro-generator that uses resource-names will error if the imports aren't correct; but it'd be better to catch this earlier.

Reply:
done-zo

@chrisdunelm, what do you mean by "check that all resource-name dependencies have been imported properly"?

Reply:
If a proto file only needs importing for a resource_reference then protoc won't be able to check that the correct import is present because protoc doesn't know about resource_reference annotations. So the linter should check this instead.